src/pip/_internal/download.py

```diff
@@ -491,61 +503,17 @@ class SafeFileCache(FileCache):
     def __init__(self, *args, **kwargs):
         super(SafeFileCache, self).__init__(*args, **kwargs)
 
-        # Check to ensure that the directory containing our cache directory
-        # is owned by the user current executing pip. If it does not exist
-        # we will check the parent directory until we find one that does exist.
-        # If it is not owned by the user executing pip then we will disable
-        # the cache and log a warning.
-        if not check_path_owner(self.directory):
-            logger.warning(
-                "The directory '%s' or its parent directory is not owned by "
-                "the current user and the cache has been disabled. Please "
-                "check the permissions and owner of that directory. If "
-                "executing pip with sudo, you may want sudo's -H flag.",
-                self.directory,
-            )
-
-            # Set our directory to None to disable the Cache
-            self.directory = None
-
     def get(self, *args, **kwargs):
-        # If we don't have a directory, then the cache should be a no-op.
-        if self.directory is None:
-            return
-
-        try:
+        with suppressed_cache_errors():
             return super(SafeFileCache, self).get(*args, **kwargs)
```

Comment:
Since `self.directory is None` had special handling in the original code, I would add a code comment here saying what `FileCache.get()` does if `self.directory` is `None` (and similarly for the other places that had that handling). Looking at the code, it looks like `TypeError` would be raised (but I could be wrong), but are the tests still passing? https://github.com/pypa/pip/blob/5a1b97d572937a3b671e998fc1d9a1ef1730593f/src/pip/_vendor/cachecontrol/caches/file_cache.py#L106

Reply:
In general I would rather throw a `ValueError` in the constructor. In this case given the single non-test use (when the input `directory` is guaranteed to be truthy [here](https://github.com/pypa/pip/pull/6845/files#diff-34481ab2fefef95e8820a46e86004ae7R581)) and the upcoming makeover, I would prefer to leave it as-is.

Reply:
If we know self.directory will always be non-None, then I would add that assertion in the constructor. (That’s also a change you’re making in this code — that it won’t ever be None, so it would be good to reflect that.) Without that, there’s no way to tell by looking at it because you have to do a search of the whole code base.

Reply:
Addressed [here](https://github.com/pypa/pip/pull/6845/files#diff-34481ab2fefef95e8820a46e86004ae7R503-R505).